addons/mail/migrations/12.0.1.0/post-migration.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2018 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+from psycopg2.extensions import AsIs
+
+
+def fill_mail_blacklist_res_partner(cr):
+    cr.execute(
+        """
+        INSERT INTO mail_blacklist (email, active,
+            create_uid, create_date, write_uid, write_date)
+        SELECT email, BOOL_OR(active), MAX(create_uid), MAX(create_date),
+            MAX(write_uid), MAX(write_date)
+        FROM res_partner
+        WHERE %s AND email IS NOT NULL
+            AND email NOT IN (SELECT email FROM mail_blacklist)
```

Comment:
You don't need this part as you are already grouping by email. That's the main improvement why why group.